tests/Builder/Stage/MergeStageTest.php

```diff
@@ -0,0 +1,188 @@
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::MergeMergeResultsFromMultipleCollections, $pipeline);
+    }
+
+    public function testOnDemandMaterializedViewInitialCreation(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::group(
+                _id: object(
+                    fiscal_year: Expression::stringFieldPath('fiscal_year'),
+                    dept: Expression::stringFieldPath('dept'),
+                ),
+                salaries: Accumulator::sum(
+                    Expression::numberFieldPath('salary'),
+                ),
+            ),
+            Stage::merge(
+                into: object(
+                    db: 'reporting',
+                    coll: 'budgets',
+                ),
+                on: '_id',
+                whenMatched: 'replace',
+                whenNotMatched: 'insert',
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::MergeOnDemandMaterializedViewInitialCreation, $pipeline);
+    }
+
+    public function testOnDemandMaterializedViewUpdateReplaceData(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::match(
+                fiscal_year: Query::gte(2019),
+            ),
+            Stage::group(
+                _id: object(
+                    fiscal_year: Expression::stringFieldPath('fiscal_year'),
+                    dept: Expression::stringFieldPath('dept'),
+                ),
+                salaries: Accumulator::sum(
+                    Expression::numberFieldPath('salary'),
+                ),
+            ),
+            Stage::merge(
+                into: object(
+                    db: 'reporting',
+                    coll: 'budgets',
+                ),
+                on: '_id',
+                whenMatched: 'replace',
+                whenNotMatched: 'insert',
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::MergeOnDemandMaterializedViewUpdateReplaceData, $pipeline);
+    }
+
+    public function testOnlyInsertNewData(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::match(
+                fiscal_year: 2019,
+            ),
+            Stage::group(
+                _id: object(
+                    fiscal_year: Expression::stringFieldPath('fiscal_year'),
+                    dept: Expression::stringFieldPath('dept'),
+                ),
+                employees: Accumulator::push(
+                    Expression::numberFieldPath('employee'),
+                ),
+            ),
+            Stage::project(
+                _id: 0,
+                dept: Expression::fieldPath('_id.dept'),
+                fiscal_year: Expression::fieldPath('_id.fiscal_year'),
+                employees: 1,
+            ),
+            Stage::merge(
+                into: object(
+                    db: 'reporting',
+                    coll: 'orgArchive',
+                ),
+                on: ['dept', 'fiscal_year'],
```

Comment:
The `StringFieldPath` is not accepted where `string` is expected.
<img width="1071" alt="image" src="https://github.com/mongodb/mongo-php-builder/assets/400034/1f2d3b5b-857a-4e4e-a40c-5a2cc9bf28ed">

But phpdoc types are not expressive enough to restrict an array of FieldPath.

<img width="886" alt="image" src="https://github.com/mongodb/mongo-php-builder/assets/400034/2f6a14ed-2edb-45d5-b0c3-3bd62ffc4000">

Also @alcaeus if we want to be able to alias field names, we will need objects similar to FieldPath instead of only raw strings here.

[PHPLIB-1382](https://jira.mongodb.org/browse/PHPLIB-1382)

Reply:
> if we want to be able to alias field names, we will need objects similar to FieldPath instead of only raw strings here.

Yes, I was thinking the same when I saw the code. We can add that when we look at integrating the builder into ODM itself.

Reply:
@GromNaN feel free to resolve this.